src/routes/transactions/mappers/common/swap-order.mapper.spec.ts

```diff
@@ -380,4 +382,61 @@ describe('Swap Order Mapper tests', () => {
       ).toHaveBeenCalledWith(transaction, dataSize, chainId, null, null);
     },
   );
+
+  it.each([
+    // [executedAmount, amount, expectedFilledPercentage]
+    [1000, 1000, '100.00'],
+    [0, 1000, '0.00'],
+    [500, 1000, '50.00'],
+    [350, 1050, '33.33'],
+  ])(
+    'should calculate the filled percentage correctly for buy orders',
+    async (executedAmount, amount, expected) => {
+      const chainId = faker.string.numeric();
+      const transaction = multisigTransactionBuilder().build();
+      const buyToken = tokenBuilder().with('decimals', 0).build();
+      const sellToken = tokenBuilder().build();
+      let order = orderBuilder()
+        .with(
+          'status',
+          faker.helpers.arrayElement([
+            'open',
+            'fulfilled',
+            'cancelled',
+            'expired',
+          ]),
+        )
+        .build();
+      if (order.kind === 'buy') {
+        order = {
+          ...order,
+          executedBuyAmount: BigInt(executedAmount),
+          buyAmount: BigInt(amount),
+        };
+      } else if (order.kind === 'sell') {
+        order = {
+          ...order,
+          executedSellAmount: BigInt(executedAmount),
+          sellAmount: BigInt(amount),
```

Comment:
Nit question: is there a reason we spread the object and not just assign the values directly, e.g.
```suggestion
        order.executedBuyAmount = BigInt(executedAmount)
        order.buyAmount = BigInt(amount)
      } else if (order.kind === 'sell') {
        order.executedSellAmount = BigInt(executedAmount)
        order.sellAmount = BigInt(amount)
```

Reply:
No reason. I'm used to field immutability by default so I created an object with a "copy" of the other one while overriding some values. Fixed here [7c939f5](https://github.com/safe-global/safe-client-gateway/pull/1338/commits/7c939f5a5ebab7af689bd6e372923f188ce5a215)